### PR TITLE
Mail Listener v2 - handle raw unicode chars

### PR DESCRIPTION
**Pull request by Itay4**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/37043

## Description
fix for:
```
Failed to execute fetch-incidents command. Error: Failed to parse mail data
Exception message is [Traceback (most recent call last):
File "<string>", line 7767, in __init__
File "/usr/local/lib/python3.9/site-packages/mailparser/mailparser.py", line 118, in parse_from_bytes
return MailParser.from_bytes(bt)
File "/usr/local/lib/python3.9/site-packages/mailparser/mailparser.py", line 241, in from_bytes
return cls(message)
File "/usr/local/lib/python3.9/site-packages/mailparser/mailparser.py", line 138, in __init__
self.parse()
File "/usr/local/lib/python3.9/site-packages/mailparser/mailparser.py", line 446, in parse
payload = payload.decode('raw-unicode-escape')
UnicodeDecodeError: 'rawunicodeescape' codec can't decode bytes in position 2019-2020: truncated \UXXXXXXXX escape
```

## Does it break backward compatibility?
   - [x] No

## Must have
- [ ] Tests - could not reproduce the issue to write a test for it

### CODE REVIEW

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 1]
```diff
@@ -21,6 +21,7 @@ def __init__(self, message_bytes: bytes, include_raw_body: bool, save_file: bool
             include_raw_body: Whether to include the raw body of the mail in the incident's body
             save_file: Whether to save the .eml file of the incident's mail
         """
+        message_bytes = message_bytes.replace(b'\\', b'/')
```
Comment (glicht) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:24]: need to see if we can do something better here.

**Comment by Itay4** [comment]
see failing unit test in https://app.circleci.com/pipelines/github/demisto/content/87361/workflows/43864d36-85f8-4119-a7e9-12cc5a4b4a3a/jobs/361481

**Review by ShahafBenYakir — APPROVED** [review]

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 2]
```diff
@@ -21,7 +21,12 @@ def __init__(self, message_bytes: bytes, include_raw_body: bool, save_file: bool
             include_raw_body: Whether to include the raw body of the mail in the incident's body
             save_file: Whether to save the .eml file of the incident's mail
         """
-        email_object = parse_from_bytes(message_bytes)
+        try:
+            email_object = parse_from_bytes(message_bytes)
+        except UnicodeDecodeError as e:
+            demisto.debug(f'Failed parsing mail from bytes: [{e}]. Will replace backslash and try to parse again')
+            message_bytes = message_bytes.replace(b'\\', b'/')
```
Comment (glicht) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:28]: replace only \U

**Comment by Itay4** [comment]
@glicht i've made the requested changes, please review

**Review by glicht — APPROVED** [review]

File: Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py [hunk 3]
```diff
@@ -21,9 +21,14 @@ def __init__(self, message_bytes: bytes, include_raw_body: bool, save_file: bool
             include_raw_body: Whether to include the raw body of the mail in the incident's body
             save_file: Whether to save the .eml file of the incident's mail
         """
-        email_object = parse_from_bytes(message_bytes)
-        self.id = id_
         self.mail_bytes = message_bytes
+        try:
+            email_object = parse_from_bytes(message_bytes)
+        except UnicodeDecodeError as e:
+            demisto.debug(f'Failed parsing mail from bytes: [{e}]. Will replace backslash and try to parse again')
```
Comment (glicht) [inline Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py:28]: One small comment. Let's log this in info and also add the backtrace.

**Review by yaakovi — APPROVED** [review]

**Pull request merged**